drivers/flash/nrf_qspi_nor.c

```diff
@@ -410,6 +433,74 @@ static int qspi_wait_while_writing(const struct device *dev)
 	return (ret < 0) ? ret : 0;
 }
 
+static int qspi_wrsrX(const struct device *dev, uint8_t sr_val, uint8_t sr_num)
+{
+	int ret = 0;
+	uint8_t opcode = SPI_NOR_CMD_WRSR;
+	uint8_t length = 1;
+	uint8_t sr_array[2] = {0};
+
+	if (sr_num > 2 || sr_num == 0) {
+		return -EINVAL;
+	}
+
+	if (sr_num == 1) {
+		sr_array[0] = sr_val;
+		if (INST_0_QER == JESD216_DW15_QER_S2B1v1) {
+			/* Writing sr1 clears sr2. Need to read/modify/write both. */
+			ret = qspi_rdsr2(dev);
+			if (ret < 0) {
+				LOG_ERR("RDSR for WRSR failed: %d", ret);
+				return ret;
+			}
+			sr_array[1] = ret;
+			length = 2;
+		}
+	} else if (INST_0_QER == JESD216_DW15_QER_S2B1v1
+			|| INST_0_QER == JESD216_DW15_QER_S2B1v4
+			|| INST_0_QER == JESD216_DW15_QER_S2B1v5) {
+		/* Writing sr2 requires writing sr1 as well.
+		 * Uses standard WRSR opcode
+		 */
+		sr_array[1] = sr_val;
+		ret = qspi_rdsr(dev);
+		if (ret < 0) {
+			LOG_ERR("RDSR for WRSR failed: %d", ret);
+			return ret;
+		}
+		sr_array[0] = ret;
+		length = 2;
+	} else if (INST_0_QER == JESD216_DW15_QER_S2B1v6) {
+		/* Writing sr2 uses a dedicated WRSR2 command */
+		sr_array[0] = sr_val;
+		opcode = SPI_NOR_CMD_WRSR2;
+	} else {
```

Comment:
all these "if/else" are based on constant comparison (because driver is not multi-instance and definitions are used directly -- ugly). So I think: just use `#ifdef` to select the right path to compile, or, use a macro to check for equality (like e.g. IS_ENABLED is used for constant checks).

Reply:
The first one to that, line 447 is against `sr_num` which is function parameter.

Reply:
I think I got something that's not uglier than it was, but moves these to compile times checks. @gmarull @de-nordic can you review?